### PR TITLE
Implementation of functionality with bloc

### PR DESCRIPTION
**Pull request by Juliana-Dimeglio**
Implementation of the dice roller functionalities using bloc

https://user-images.githubusercontent.com/85245831/122574956-990c8000-d026-11eb-828f-14e456e225fd.mp4

### CODE REVIEW

File: lib/scr/bloc/dice_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'i_dice_bloc.dart';
+
+class DiceBloc extends IDiceBloc {
+  final int _numberOfDice = 4;
```
Comment (FranciscoGaraventa) [inline lib/scr/bloc/dice_bloc.dart]: The number of dices that you used can be a constant

File: lib/scr/bloc/dice_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  List<int> defaultValue() {
+    var sum = 0;
+    List<int> _listAux = [];
+    for (var i = 0; i < numberOfDice; i++) {
+      _listAux.add(Random().nextInt(6) + 1);
+      sum += _listAux[i];
+    }
+    _scoreValue.sink.add(sum);
+    return _listAux;
+  }
```
Comment (FranciscoGaraventa) [inline lib/scr/bloc/dice_bloc.dart]: Here you are doing similar code in both functions (roll and defaultValue),try to separate the portion of code that repeats into a separate function and then call it in each method where it is used.

File: lib/scr/bloc/dice_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    var sum = 0;
+    for (var i = 0; i < numberOfDice; i++) {
+      _listAux.add(Random().nextInt(6) + 1);
+      sum += _listAux[i];
+    }
+    _streamController.sink.add(_listAux);
+    _scoreValue.sink.add(sum);
+  }
+
+  List<int> defaultValue() {
```
Comment (FranciscoGaraventa) [inline lib/scr/bloc/dice_bloc.dart]: This function can be named different like initialDiceValues ​​or something like that

File: lib/scr/screen_dice_roller.dart [hunk 4]
```diff
@@ -33,26 +36,70 @@ class _ScreenDiceRollerState extends State<ScreenDiceRoller> {
...
-              child: Image.asset(
-                "assets/images/Dice_${(index % Constants.modNumber) + 1}.png",
-                height: Constants.imageHeight,
-              ),
+        body: StreamBuilder(
+          stream: widget.bloc.diceStream,
+          initialData: widget.bloc.defaultValue(),
+          builder: (context, snapshot) {
+            return GridView.count(
+              crossAxisCount: 2,
```
Comment (FranciscoGaraventa) [inline lib/scr/screen_dice_roller.dart]: This can be a constant

File: lib/scr/screen_dice_roller.dart [hunk 5]
```diff
@@ -33,26 +36,70 @@ class _ScreenDiceRollerState extends State<ScreenDiceRoller> {
...
+              children: List.generate(widget.bloc.numberOfDice, (index) {
+                return Padding(
+                  padding: const EdgeInsets.all(Constants.paddingImage),
```
Comment (FranciscoGaraventa) [inline lib/scr/screen_dice_roller.dart]: You missed a trailing comma here

File: lib/scr/screen_dice_roller.dart [hunk 6]
```diff
@@ -33,26 +36,70 @@ class _ScreenDiceRollerState extends State<ScreenDiceRoller> {
...
+                ),
+                child: Padding(
+                  padding: const EdgeInsets.symmetric(
+                    vertical: Constants.paddingVertical,
+                    horizontal: Constants.paddingHorizontal,
+                  ),
+                  child: Text(
+                    "Roll the dice!",
+                    style: TextStyle(
+                        fontSize: Constants.fontSizeText, color: Colors.black),
```
Comment (FranciscoGaraventa) [inline lib/scr/screen_dice_roller.dart]: Missed a trailing comma here

File: lib/scr/screen_dice_roller.dart [hunk 7]
```diff
@@ -33,26 +36,70 @@ class _ScreenDiceRollerState extends State<ScreenDiceRoller> {
...
         bottomNavigationBar: Padding(
-          padding: const EdgeInsets.only(bottom: Constants.paddingText),
-          child: Text(
-            "Press a dice!\nYour score: 10",
-            style: TextStyle(
-              fontSize: Constants.fontSizeText,
-              fontStyle: FontStyle.italic,
-            ),
-            textAlign: TextAlign.center,
+          padding: const EdgeInsets.only(bottom: Constants.paddingBottomNav),
```
Comment (MarcosCGlobant) [inline lib/scr/screen_dice_roller.dart]: Missed a trailing comma here

File: lib/utils/constants_file.dart [hunk 8]
```diff
@@ -1,8 +1,9 @@
...
   static const double fontSizeTitle = 25;
-  static const double fontSizeText = 40;
+  static const double fontSizeText = 30;
   static const int numberOfDice = 4;
-  static const double paddingText = 70;
+  static const double paddingImage = 20;
+  static const double paddingBottomNav = 40;
+  static const double paddingHorizontal = 20;
+  static const double paddingVertical = 10;
 }
```
Comment (MarcosCGlobant) [inline lib/utils/constants_file.dart]: You can remove the "_file" from the name of the file, with the first part it's enough

Comment (Juliana-Dimeglio) [reply lib/utils/constants_file.dart]: Fixed

File: lib/scr/screen_dice_roller.dart (hunk 7 above)
Comment (Juliana-Dimeglio) [reply lib/scr/screen_dice_roller.dart]: Fixed

File: lib/scr/screen_dice_roller.dart (hunk 6 above)
Comment (Juliana-Dimeglio) [reply lib/scr/screen_dice_roller.dart]: Fixed

File: lib/scr/screen_dice_roller.dart (hunk 5 above)
Comment (Juliana-Dimeglio) [reply lib/scr/screen_dice_roller.dart]: Fixed

File: lib/scr/screen_dice_roller.dart (hunk 4 above)
Comment (Juliana-Dimeglio) [reply lib/scr/screen_dice_roller.dart]: Fixed

File: lib/scr/bloc/dice_bloc.dart (hunk 3 above)
Comment (Juliana-Dimeglio) [reply lib/scr/bloc/dice_bloc.dart]: Fixed

File: lib/scr/bloc/dice_bloc.dart (hunk 2 above)
Comment (Juliana-Dimeglio) [reply lib/scr/bloc/dice_bloc.dart]: Fixed

File: lib/scr/bloc/dice_bloc.dart (hunk 1 above)
Comment (Juliana-Dimeglio) [reply lib/scr/bloc/dice_bloc.dart]: Fixed

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Pull request merged**